### PR TITLE
EXSWHTEC-105 - Implement tests for hipModuleOccupancyMaxActiveBlocksPerMultiprocessor and hipModuleOccupancyMaxPotentialBlockSize APIs

### PR DESCRIPTION
**Pull request by nives-vukovic**
- Add helper file occupancy_common.hh with parameterized templates
- Implement tests for hipModuleOccupancyMaxActiveBlocksPerMultiprocessor
- Implement tests for hipModuleOccupancyMaxPotentialBlockSize
- Modify CMakeLists.txt to include compilation of simple_kernel.code

### CODE REVIEW

File: catch/unit/occupancy/hipModuleOccupancyMaxActiveBlocksPerMultiprocessorWithFlags.cc [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+
+  // Common negative tests
+  MaxActiveBlocksPerMultiprocessorNegative(
+      [&function](int* numBlocks, int blockSize, size_t dynSharedMemPerBlk) {
+        return hipModuleOccupancyMaxActiveBlocksPerMultiprocessorWithFlags(
+            numBlocks, function, blockSize, dynSharedMemPerBlk, hipOccupancyDefault);
+      },
+      blockSize);
+
+#if HT_NVIDIA // EXSWHTEC-219
```
Comment (chrispaquot) [inline catch/unit/occupancy/hipModuleOccupancyMaxActiveBlocksPerMultiprocessorWithFlags.cc]: again here, remove this and add the test case to json file

Comment (nives-vukovic) [reply catch/unit/occupancy/hipModuleOccupancyMaxActiveBlocksPerMultiprocessorWithFlags.cc]: @chrispaquot  It has been removed and test has been disabled using json files

**Review by chrispaquot — APPROVED** [review]

**Comment by rorake** [comment]
@nives-vukovic 
Will you please resolve the merge conflicts for this PR?

**Comment by nives-vukovic** [comment]
@rorake Conflicts are resolved.

**Comment by mangupta** [comment]
PR has been merged into develop branch via commit bb791d8e8680f9fc15ca8ae561ba39623c47a581

**Pull request closed without merging**